### PR TITLE
Fix `strptime` and `UnixTime` for windows, for #43

### PR DESCRIPTION
**Pull request by sighingnow**
This patch imports a portable implementation of `strptime` from [strptime.c](https://github.com/Alexpux/MINGW-packages/blob/master/mingw-w64-libkml/strptime.c), and fixes windows related issues of the `Storable` instance of `UnixTime`. I have left some notes in changeset.

### CODE REVIEW

File: cbits/conv.c [hunk 1]
```diff
@@ -32,11 +33,43 @@ void init_locale() {
     static int initialized = 0;
     if (initialized == 0) {
         setlocale(LC_TIME, "C");
-	initialized == 1;
+        initialized = 1;
```
Comment (sighingnow) [inline cbits/conv.c]: I believe here is an unintended error.

**Comment by sighingnow** [comment]
Currently the tests on windows is not OK. The case `behaves like the model` and `inverses the result (2)` fail to pass the tests due to the `%s` and `%z` flag of `strftime`.

**Comment by kazu-yamamoto** [comment]
I set up AppVeyor: https://ci.appveyor.com/project/kazu-yamamoto/unix-time
But not matured yet.

**Comment by sighingnow** [comment]
@kazu-yamamoto I have looked at the appveyor failure. It seems an out-of-memory error and I guess we need to use 64-bit stack, see: https://github.com/sighingnow/unix-time/commit/b252cff7870c936fee05de922acb2a926220d5c3 .

After that, we have no problem with installing dependencies, then we can focus on the real failed test cases of unix-time, see: https://ci.appveyor.com/project/sighingnow/unix-time/builds/20945553

Could you please try the same change to appveyor.yaml in your repo ?

**Comment by sighingnow** [comment]
Done.

**Comment by kazu-yamamoto** [comment]
@sighingnow Are you planning to implement `%s` and `%z`?

**Comment by kazu-yamamoto** [comment]
OK. Thanks.

**Comment by sighingnow** [comment]
@kazu-yamamoto I have imported `strftime` and `strptime` from FreeBSD, and fixed the windows-specific problems. The license should ok since we are also use BDS-3 license.

Local `stack test` is green now.

**Comment by sighingnow** [comment]
Now both travis and appveyor are green!

**Comment by kazu-yamamoto** [comment]
Merged and released.
Thank you for your great contribution!

**Pull request merged**